vms/evm.ts

```diff
@@ -1,199 +1,273 @@
+    MAX_PRIORITY_FEE: string
+    MAX_FEE: string
+    RECALIBRATE: number
+    hasNonce: Map<string, number | undefined>
+    pendingTxNonces: Set<unknown>
+    hasError: Map<string, string | undefined>
+    nonce: number
+    balance: any
+    isFetched: boolean
+    isUpdating: boolean
+    recalibrate: boolean
+    waitingForRecalibration: boolean
+    waitArr: any[]
+    queue: any[]
+    error: boolean
+    log: Log
+    contracts: any
+
+    constructor(config: ChainType, PK: string | undefined) {
+        this.web3 = new Web3(config.RPC)
+        this.account = this.web3.eth.accounts.privateKeyToAccount(PK)
+        this.contracts = new Map()
+
+        this.NAME = config.NAME
+        this.DRIP_AMOUNT = (new BN(config.DRIP_AMOUNT)).mul(new BN(1e9))
+        this.MAX_PRIORITY_FEE = config.MAX_PRIORITY_FEE
+        this.MAX_FEE = config.MAX_FEE
+        this.RECALIBRATE = config.RECALIBRATE || 30
+        this.LEGACY = false
 
         this.log = new Log(this.NAME)
 
-        this.hasNonce = new Map();
-        this.hasError = new Map();
-        this.pendingTxNonces = new Set();
+        this.hasNonce = new Map()
+        this.hasError = new Map()
+        this.pendingTxNonces = new Set()
 
-        this.nonce = -1;
-        this.balance = new BN(0);
+        this.nonce = -1
+        this.balance = new BN(0)
 
-        this.isFetched = false;
-        this.isUpdating = false;
-        this.recalibrate = false;
-        this.waitingForRecalibration = false;
+        this.isFetched = false
+        this.isUpdating = false
+        this.recalibrate = false
+        this.waitingForRecalibration = false
 
-        this.waitArr = [];
-        this.queue = [];
+        this.waitArr = []
+        this.queue = []
 
-        this.error = false;
+        this.error = false
 
-        this.setupTransactionType();
-        this.recalibrateNonceAndBalance();
+        this.setupTransactionType()
+        this.recalibrateNonceAndBalance()
 
         setInterval(() => {
-            this.recalibrateNonceAndBalance();
-        }, this.RECALIBRATE * 1000);
+            this.recalibrateNonceAndBalance()
+        }, this.RECALIBRATE * 1000)
     }
 
-    async setupTransactionType() {
+    // Setup Legacy or EIP1559 transaction type
+    async setupTransactionType(): Promise<void> {
         try {
             const baseFee = (await this.web3.eth.getBlock('latest')).baseFeePerGas
             if(baseFee == undefined) {
-                this.LEGACY = true;
+                this.LEGACY = true
             }
-            this.error = false;
+            this.error = false
         } catch(err: any) {
-            this.error = true;
-            this.log.error(err.message);
+            this.error = true
+            this.log.error(err.message)
         }
     }
 
-    async sendToken(receiver: string, cb: (param: SendTokenResponse) => void): Promise<void> {
+    // Function to issue transfer transaction. For ERC20 transfers, id will be string
```

Comment:
nit: it

Reply:
Added more explanation. Actually, it's `id` representing ERC20 token ID as mentioned in the config file.